Import python libraries.

In [1]:
import pandas as pd
import numpy as np

Read and preview the data.

In [2]:
data = pd.read_csv('kiva_loans.csv')
data.head()

id  funded_amount  loan_amount             activity          sector  \
0  653051          300.0        300.0  Fruits & Vegetables            Food   
1  653053          575.0        575.0             Rickshaw  Transportation   
2  653068          150.0        150.0       Transportation  Transportation   
3  653063          200.0        200.0           Embroidery            Arts   
4  653084          400.0        400.0           Milk Sales            Food   

                                                 use country_code   country  \
0            To buy seasonal, fresh fruits to sell.            PK  Pakistan   
1  to repair and maintain the auto rickshaw used ...           PK  Pakistan   
2  To repair their old cycle-van and buy another ...           IN     India   
3  to purchase an embroidery machine and a variet...           PK  Pakistan   
4                           to purchase one buffalo.           PK  Pakistan   

         region currency  partner_id                posted_time  \
0        Lahore      PKR       247.0  2014-01-01 06:12:39+00:00   
1        Lahore      PKR       247.0  2014-01-01 06:51:08+00:00   
2     Maynaguri      INR       334.0  2014-01-01 09:58:07+00:00   
3        Lahore      PKR       247.0  2014-01-01 08:03:11+00:00   
4  Abdul Hakeem      PKR       245.0  2014-01-01 11:53:19+00:00   

              disbursed_time                funded_time  term_in_months  \
0  2013-12-17 08:00:00+00:00  2014-01-02 10:06:32+00:00            12.0   
1  2013-12-17 08:00:00+00:00  2014-01-02 09:17:23+00:00            11.0   
2  2013-12-17 08:00:00+00:00  2014-01-01 16:01:36+00:00            43.0   
3  2013-12-24 08:00:00+00:00  2014-01-01 13:00:00+00:00            11.0   
4  2013-12-17 08:00:00+00:00  2014-01-01 19:18:51+00:00            14.0   

   lender_count                          tags borrower_genders  \
0            12                           NaN           female   
1            14                           NaN   female, female   
2             6  user_favorite, user_favorite           female   
3             8                           NaN           female   
4            16                           NaN           female   

  repayment_interval        date  
0          irregular  2014-01-01  
1          irregular  2014-01-01  
2             bullet  2014-01-01  
3          irregular  2014-01-01  
4            monthly  2014-01-01

Look at data columns, null values and data types.

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 20 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   funded_amount       671205 non-null  float64
 2   loan_amount         671205 non-null  float64
 3   activity            671205 non-null  object 
 4   sector              671205 non-null  object 
 5   use                 666973 non-null  object 
 6   country_code        671197 non-null  object 
 7   country             671205 non-null  object 
 8   region              614405 non-null  object 
 9   currency            671205 non-null  object 
 10  partner_id          657698 non-null  float64
 11  posted_time         671205 non-null  object 
 12  disbursed_time      668809 non-null  object 
 13  funded_time         622874 non-null  object 
 14  term_in_months      671205 non-null  float64
 15  lender_count        671205 non-nul

I will drop the following columns because they do not any information that will help in the analysis of Kiva borrowers.

In [4]:
#drop the following columns because they do not have a direct relationship with the borrowers of Kiva.
data.drop(['funded_amount', 'activity', 'use', 'country_code', 'region', 'currency', 'partner_id', 'posted_time', 'disbursed_time', 'funded_time', 'lender_count', 'tags'], axis=1, inplace=True)

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 671205 entries, 0 to 671204
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   id                  671205 non-null  int64  
 1   loan_amount         671205 non-null  float64
 2   sector              671205 non-null  object 
 3   country             671205 non-null  object 
 4   term_in_months      671205 non-null  float64
 5   borrower_genders    666984 non-null  object 
 6   repayment_interval  671205 non-null  object 
 7   date                671205 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 41.0+ MB


In [6]:
data.sample(10)

id  loan_amount          sector      country  term_in_months  \
491842  1151487        750.0       Education   Tajikistan            20.0   
31287    684030       1500.0  Transportation      Bolivia            20.0   
668925  1333804        200.0        Services        Kenya            11.0   
43779    696217       2225.0            Food         Mali             7.0   
613036  1275019        600.0            Food  El Salvador            17.0   
453041  1112540        550.0     Agriculture  Philippines            13.0   
663123  1325598        500.0          Retail     Pakistan            14.0   
245137   892992       1150.0     Agriculture  El Salvador            14.0   
178209   826932        575.0     Agriculture     Cambodia            14.0   
630931  1293497        675.0     Agriculture     Colombia            20.0   

                                         borrower_genders repayment_interval  \
491842                                             female            monthly   
31287                                                male            monthly   
668925                                             female          irregular   
43779   female, female, female, female, female, female...            monthly   
613036                                             female            monthly   
453041                                             female          irregular   
663123                                             female            monthly   
245137                                             female             bullet   
178209                     female, female, female, female            monthly   
630931                                             female             bullet   

              date  
491842  2016-09-16  
31287   2014-03-14  
668925  2017-07-05  
43779   2014-04-10  
613036  2017-04-06  
453041  2016-07-07  
663123  2017-06-22  
245137  2015-05-27  
178209  2015-01-12  
630931  2017-05-05

I'll change the date column to a datetime data type for easy time analysis.

In [7]:
data['date'] = pd.to_datetime(data['date'], format='%Y-%m-%d')

I'll use isnull() method and chain it with sum() to find the total number of nulls in each column in the DataFrame

In [8]:
data.isnull().sum()

id                       0
loan_amount              0
sector                   0
country                  0
term_in_months           0
borrower_genders      4221
repayment_interval       0
date                     0
dtype: int64

In [9]:
data.shape

(671205, 8)

I'll drop the rows with missing values, since 4221 is not such a significant number to do away with in my DataFrame.

In [10]:
#drop any row with a missing value
data.dropna(axis=0, how='any', inplace=True)

In [11]:
data.shape

(666984, 8)

I have noticed some anomalies in the gender column that could have occurred during data collection, so I'll clean it up.

In [12]:
data['borrower_genders'].sample(10)

355990                    female
269203            female, female
216654                    female
135012                    female
50718     female, female, female
172710                    female
579792                    female
88044                     female
122181                      male
108767                    female
Name: borrower_genders, dtype: object

I'll create a new column, of boolean type, to contain strings that contains female. True contains female, and False will hence contain male.

In [13]:
data['female_gender'] = data['borrower_genders'].str.contains('female')

I'll create a gender column that encodes the above classification into 'F' and 'M'.

In [14]:
data.loc[data['female_gender'] == True, 'gender'] = 'F'
data.loc[data['female_gender'] == False, 'gender'] ='M'

I will drop the following columns since I now have a clean gender column. Using the drop() method, with axis and inplace arguments.

In [15]:
data.drop(['borrower_genders', 'female_gender'], axis=1, inplace=True)

Now let's look at unique values and their counts in the gender column, using value_counts() method.

In [16]:
print(data['gender'].value_counts())

F    528461
M    138523
Name: gender, dtype: int64


I will convert the gender column into a category.

In [17]:
category = ['F', 'M']
cat_dtype = pd.api.types.CategoricalDtype(categories=category, ordered=True)
data['gender'] = data['gender'].astype(cat_dtype)

print(data.gender.sample(5))

60157     F
80458     M
440130    F
3690      F
610442    F
Name: gender, dtype: category
Categories (2, object): [F < M]


Let's look at the repayment_interval column.
Use value_counts() to find counts of unique values in the column.

In [18]:
print(data['repayment_interval'].value_counts())

monthly      340171
irregular    256573
bullet        69638
weekly          602
Name: repayment_interval, dtype: int64


bullet could mean daily, so I'll change it.

In [19]:
data.loc[data['repayment_interval'] == 'bullet', 'repayment_interval'] = 'daily'

I'll change the repayment_interval column to a category too. Directly, using astype() method, with category as an argument.

In [20]:
data['repayment_interval'] = data['repayment_interval'].astype('category')
print(data['repayment_interval'].dtype)
print(data.repayment_interval.value_counts())

category
monthly      340171
irregular    256573
daily         69638
weekly          602
Name: repayment_interval, dtype: int64


Let's look at the country column

In [21]:
data.country.head()

0    Pakistan
1    Pakistan
2       India
3    Pakistan
4    Pakistan
Name: country, dtype: object

In [22]:
print(data.country.unique())
print(data.country.nunique())

['Pakistan' 'India' 'Kenya' 'Nicaragua' 'El Salvador' 'Tanzania'
 'Philippines' 'Peru' 'Senegal' 'Cambodia' 'Liberia' 'Vietnam' 'Iraq'
 'Honduras' 'Palestine' 'Mongolia' 'United States' 'Mali' 'Colombia'
 'Tajikistan' 'Guatemala' 'Ecuador' 'Bolivia' 'Yemen' 'Ghana'
 'Sierra Leone' 'Haiti' 'Chile' 'Jordan' 'Uganda' 'Burundi' 'Burkina Faso'
 'Timor-Leste' 'Indonesia' 'Georgia' 'Ukraine' 'Kosovo' 'Albania'
 'The Democratic Republic of the Congo' 'Costa Rica' 'Somalia' 'Zimbabwe'
 'Cameroon' 'Turkey' 'Azerbaijan' 'Dominican Republic' 'Brazil' 'Mexico'
 'Kyrgyzstan' 'Armenia' 'Paraguay' 'Lebanon' 'Samoa' 'Israel' 'Rwanda'
 'Zambia' 'Nepal' 'Congo' 'Mozambique' 'South Africa' 'Togo' 'Benin'
 'Belize' 'Suriname' 'Thailand' 'Nigeria' 'Mauritania' 'Vanuatu' 'Panama'
 'Virgin Islands' 'Saint Vincent and the Grenadines'
 "Lao People's Democratic Republic" 'Myanmar (Burma)' 'Malawi' 'Moldova'
 'South Sudan' 'Solomon Islands' 'China' 'Egypt' 'Guam' 'Afghanistan'
 'Madagascar' 'Namibia' 'Puerto Rico

Let's also look at the sector column

In [23]:
print(data.sector.unique())
print(data.sector.nunique())
data.sector.value_counts()

['Food' 'Transportation' 'Arts' 'Services' 'Agriculture' 'Manufacturing'
 'Wholesale' 'Retail' 'Clothing' 'Construction' 'Health' 'Education'
 'Personal Use' 'Housing' 'Entertainment']
15


Agriculture       179224
Food              135749
Retail            123758
Services           44699
Personal Use       36247
Housing            33571
Clothing           32480
Education          30837
Transportation     15455
Arts               11994
Health              9172
Construction        6194
Manufacturing       6158
Entertainment        820
Wholesale            626
Name: sector, dtype: int64

Because Food and Agriculture are directly related, I have changed them to one: Agriculture

In [24]:
data.loc[data['sector'] == 'Food', 'sector'] = 'Agriculture'
data.sector.unique()

array(['Agriculture', 'Transportation', 'Arts', 'Services',
       'Manufacturing', 'Wholesale', 'Retail', 'Clothing', 'Construction',
       'Health', 'Education', 'Personal Use', 'Housing', 'Entertainment'],
      dtype=object)

Check for duplicates in the whole DataFrame using the duplicated() method.

In [25]:
data.duplicated().sum()

0

### Testing hypothesis towards finding out about the borrowers in Kiva.


1.Who are they?-Gender.
2.What do they do?-Sector
3.Where do they come from?-Country
4.How long have they borrowed?-term in months
5.How do they pay? Poverty levels? -repayment intervals
6.Which country borrows the most or the least? & In which sector?


#### 1.What is the gender of most borrowers in Kiva? 

###### My hypothesis is that most borrowers in Kiva are female.

I will create variables with counts of each gender and use the total to calculate the percentage of each gender.

In [26]:
female_borrowers = data.gender[data['gender'] == 'F'].count()
male_borrowers = data.gender[data['gender'] == 'M'].count()
total_borrowers = data.shape[0]
print('The female borrowers are: {:.0%}'.format(female_borrowers /total_borrowers))
print('The male borrowers are: {:.0%}'.format(male_borrowers /total_borrowers))

The female borrowers are: 79%
The male borrowers are: 21%


##### Female borrowers, at 79% are more than male borrowers at 21%.

#### 2.In which sector are the female and male genders involved in respectively?

###### My hypothesis is that most female are in the Agriculture sector, and male in the Transportation sector.

In [27]:
female_data = data[data['gender'] == 'F']
male_data = data[data['gender'] == 'M']
female_data.head(5)
#male_data.sample(10)

id  loan_amount          sector   country  term_in_months  \
0  653051        300.0     Agriculture  Pakistan            12.0   
1  653053        575.0  Transportation  Pakistan            11.0   
2  653068        150.0  Transportation     India            43.0   
3  653063        200.0            Arts  Pakistan            11.0   
4  653084        400.0     Agriculture  Pakistan            14.0   

  repayment_interval       date gender  
0          irregular 2014-01-01      F  
1          irregular 2014-01-01      F  
2              daily 2014-01-01      F  
3          irregular 2014-01-01      F  
4            monthly 2014-01-01      F

In [28]:
female_data['sector'].value_counts()

Agriculture       250078
Retail            109244
Services           34512
Clothing           29752
Personal Use       27580
Housing            25177
Education          19008
Arts               10758
Transportation      8521
Health              6203
Manufacturing       3799
Construction        3047
Entertainment        424
Wholesale            358
Name: sector, dtype: int64

Most female borrowers are in the Agriculture sector.

In [29]:
print('The sector with the most amount of loan borrowed:', data.groupby('sector')['loan_amount'].sum().idxmax())
print('The highest loan amount borrowed per sector:{:.0f}'.format(data.groupby('sector')['loan_amount'].sum().max()))

The sector with the most amount of loan borrowed: Agriculture
The highest loan amount borrowed per sector:262627050


In [30]:
male_data['sector'].value_counts()

Agriculture       64895
Retail            14514
Education         11829
Services          10187
Personal Use       8667
Housing            8394
Transportation     6934
Construction       3147
Health             2969
Clothing           2728
Manufacturing      2359
Arts               1236
Entertainment       396
Wholesale           268
Name: sector, dtype: int64

Most male borrowers are also in the Agriculture sector.

##### Both male and female borrowers are in the agricultural sector.

#### 3. In which country do most borrowers come from?

I will use the groupby() method to group my data by country,  and the agg() method to find the value counts in the female column.

In [31]:
country_grouped = data.groupby('country').agg({'gender': ['value_counts']})
country_grouped.head(5)

gender
                   value_counts
country     gender             
Afghanistan F                 2
Albania     F              1018
            M               899
Armenia     F              5383
            M              3245

In [32]:
type(country_grouped)

pandas.core.frame.DataFrame

In [33]:
print(country_grouped['gender'].idxmin(), country_grouped['gender'].min())

value_counts    (Bhutan, F)
dtype: object value_counts    1
dtype: int64


In [34]:
print(country_grouped['gender'].idxmax(), country_grouped['gender'].max())

value_counts    (Philippines, F)
dtype: object value_counts    151984
dtype: int64


##### The highest country borrrower is Phillipines. In Phillippines, the female gender borrows the most.
##### The lowest country borrower is Bhutan. With one female borrower.

Why Phillipines? Is Kiva a Phillipines' company? Have more financial campaigns been done in Phillipines?

#### 4. How long have they borrowed?
Which is the longest borrowing gender?
Which is the longest borrowing country?

In [35]:
data.term_in_months.sample(5)
data.term_in_months.describe()

count    666984.000000
mean         13.730235
std           8.596158
min           1.000000
25%           8.000000
50%          13.000000
75%          14.000000
max         158.000000
Name: term_in_months, dtype: float64

The mean and the median are the same, so there are no outliers affecting the mean.

Although the standard deviation is too high, this means that some values are too far from the mean.

I'll group by gender to find the maximum term in months for each gender.

In [36]:
term_gender = data.groupby('gender')['term_in_months'].max()
print(term_gender)

gender
F    158.0
M    154.0
Name: term_in_months, dtype: float64


Grouping by country and then sorting in descending order to find which country has the longest borrower.

In [37]:
term_country = data.groupby('country')['term_in_months'].max()
print(term_country.sort_values(ascending=False))

country
Dominican Republic    158.0
Mexico                144.0
South Africa          141.0
Kenya                 122.0
Cambodia              120.0
                      ...  
Lesotho                12.0
Cote D'Ivoire          10.0
Congo                   8.0
Afghanistan             8.0
Namibia                 7.0
Name: term_in_months, Length: 87, dtype: float64


I will access the row with full details of the borrower with the longest and shortest term, using the .loc() access method

In [38]:
(data.loc[data['term_in_months'] == 158])

id  loan_amount     sector             country  term_in_months  \
215863  863993       6725.0  Education  Dominican Republic           158.0   

       repayment_interval       date gender  
215863          irregular 2015-03-30      F

In [39]:
data.loc[data['term_in_months']== 1]

id  loan_amount          sector country  term_in_months  \
1877    1080189        125.0        Services   Kenya             1.0   
2915    1080219        250.0        Clothing   Kenya             1.0   
4277    1080268        125.0     Agriculture   Kenya             1.0   
6772    1080311        125.0        Clothing   Kenya             1.0   
7851    1080328        125.0        Services   Kenya             1.0   
10478   1080374        125.0        Clothing   Kenya             1.0   
11319   1080391        250.0  Transportation   Kenya             1.0   
11410   1080393        250.0          Retail   Kenya             1.0   
11488   1080392        250.0     Agriculture   Kenya             1.0   
11599   1080394        250.0     Agriculture   Kenya             1.0   
11617   1080388        250.0        Services   Kenya             1.0   
11800   1080405        250.0        Services   Kenya             1.0   
11803   1080404        250.0          Retail   Kenya             1.0   
12178   1080420        250.0          Retail   Kenya             1.0   
12453   1080434        250.0        Clothing   Kenya             1.0   
14310   1080473        125.0          Retail   Kenya             1.0   
15542   1080489        125.0        Clothing   Kenya             1.0   
29016   1080675        250.0          Retail   Kenya             1.0   
29789   1080683        250.0        Clothing   Kenya             1.0   
39694   1080894        250.0    Construction   Kenya             1.0   
45830   1081076        125.0        Clothing   Kenya             1.0   
45986   1081078        125.0          Retail   Kenya             1.0   
48474   1081137        250.0     Agriculture   Kenya             1.0   
49396   1081156        250.0        Clothing   Kenya             1.0   
53098   1081263        125.0     Agriculture   Kenya             1.0   
81739   1082038        250.0  Transportation   Kenya             1.0   
82547   1082049        250.0     Agriculture   Kenya             1.0   
121850  1083671        100.0     Agriculture   Kenya             1.0   
196159  1087579         75.0     Agriculture   Kenya             1.0   
199098  1087727         75.0     Agriculture   Kenya             1.0   
208871  1088074         75.0     Agriculture   Kenya             1.0   

       repayment_interval       date gender  
1877            irregular 2014-01-08      M  
2915            irregular 2014-01-10      F  
4277            irregular 2014-01-15      M  
6772            irregular 2014-01-21      F  
7851               weekly 2014-01-23      M  
10478           irregular 2014-01-29      F  
11319           irregular 2014-01-31      F  
11410           irregular 2014-01-31      F  
11488           irregular 2014-01-31      F  
11599           irregular 2014-01-31      F  
11617           irregular 2014-01-31      M  
11800           irregular 2014-02-02      F  
11803           irregular 2014-02-02      F  
12178           irregular 2014-02-03      F  
12453           irregular 2014-02-04      F  
14310           irregular 2014-02-09      F  
15542           irregular 2014-02-11      F  
29016           irregular 2014-03-11      F  
29789           irregular 2014-03-12      F  
39694           irregular 2014-04-02      F  
45830           irregular 2014-04-16      F  
45986           irregular 2014-04-16      F  
48474           irregular 2014-04-23      M  
49396           irregular 2014-04-24      F  
53098              weekly 2014-05-03      F  
81739           irregular 2014-07-02      F  
82547           irregular 2014-07-03      F  
121850          irregular 2014-09-22      F  
196159          irregular 2015-02-20      F  
199098          irregular 2015-02-26      F  
208871          irregular 2015-03-17      F

###### I have discovered that the Dominican Republic hosts the longest borrower, while Kenya hosts the shortest borrowers!

###### This could be most likely that Kiva started its operations earlier in Dominican Republic than in Kenya.

Or that the borrower with the longest term is an outlier. Let's find out.

I'll slice my data set to find out how many customers have been borrowers for more than 5 years i.e 60 months.

In [40]:
long = data[data['term_in_months'] >60]
print(long['term_in_months'].count())

3361


Well, only 3000 out of 660000 customers have borrowed for more than 5 years. Seems insignificant enough that i might do away with them...

I'll visulaize first to see how they affect histograms and other distribution curves.

#### 5. At what intervals do they pay?
    How do they pay? Poverty levels? -repayment intervals

In [41]:
data['repayment_interval'].value_counts()

monthly      340171
irregular    256573
daily         69638
weekly          602
Name: repayment_interval, dtype: int64

In [62]:
print(data['repayment_interval'].value_counts().idxmax(), data['repayment_interval'].value_counts().max())

monthly 340171


Monthly repayments are the highest, closely followed by irregular payments.

Create a pivot table to find the count of repayment intervals by grouping by sector, using the loan amount values.

In [81]:
repayment_sector = data.pivot_table(index='sector', values='loan_amount',columns='repayment_interval', aggfunc=['count'])

repayment_sector.columns = ['daily', 'irregular', 'monthly', 'weekly']
repayment_sector.fillna(value=0, inplace=True)
repayment_sector.sort_values(by='monthly', ascending=False)

daily  irregular   monthly  weekly
sector                                              
Agriculture     47955.0   123152.0  143730.0   136.0
Retail           4764.0    73519.0   45351.0   124.0
Personal Use     4790.0     4048.0   27409.0     0.0
Services         4940.0    12518.0   27088.0   153.0
Housing           870.0     8442.0   24259.0     0.0
Education        2430.0     4331.0   24076.0     0.0
Clothing          976.0    13521.0   17845.0   138.0
Transportation    398.0     6394.0    8651.0    12.0
Health            131.0     1654.0    7374.0    13.0
Arts             1454.0     4627.0    5898.0    15.0
Construction      280.0     1545.0    4358.0    11.0
Manufacturing     441.0     2433.0    3284.0     0.0
Entertainment     155.0      226.0     439.0     0.0
Wholesale          54.0      163.0     409.0     0.0

Since most repayment intervals in general are done monthly, I have decided to sort the values by the monthly column.
I can deduce that Agriculture, Retail and Personal Use are the top three sectors in the monthly repayment interval.
This could mean that these sectors have a return period of a month.

In [86]:
data.groupby('country')['repayment_interval'].value_counts().idxmax()

('Philippines', 'irregular')

In [125]:
repayment_country = data.pivot_table(index='country', values='loan_amount',columns='repayment_interval', aggfunc=['count'])
repayment_country[repayment_country.index == 'Philippines']
repayment_country.columns = ['daily', 'irregular', 'monthly', 'weekly']
repayment_country.fillna(value=0, inplace=True)
repayment_country.head()

daily  irregular  monthly  weekly
country                                       
Afghanistan    2.0        0.0      0.0     0.0
Albania       23.0       26.0   1868.0     0.0
Armenia       43.0      170.0   8415.0     0.0
Azerbaijan     0.0       12.0   1893.0     0.0
Belize         0.0        0.0    123.0     0.0

Let's look at each sorted repayment_interval category, and identify countries that are in the top5.

In [124]:
print(repayment_country['daily'].sort_values(ascending=False).head())

country
Colombia       13382.0
Nigeria        10129.0
Kenya           9183.0
El Salvador     8083.0
India           6532.0
Name: daily, dtype: float64


In [123]:
repayment_country['weekly'].sort_values(ascending=False).head()

country
Kenya       602.0
Zimbabwe      0.0
Iraq          0.0
Georgia       0.0
Ghana         0.0
Name: weekly, dtype: float64

In [121]:
repayment_country['monthly'].sort_values(ascending=False).head()

country
Kenya          45927.0
Cambodia       34382.0
El Salvador    29329.0
Pakistan       20555.0
Tajikistan     19355.0
Name: monthly, dtype: float64

In [122]:
repayment_country['irregular'].sort_values(ascending=False).head()

country
Philippines    144899.0
Kenya           19401.0
Peru             7452.0
Samoa            7365.0
Uganda           7230.0
Name: irregular, dtype: float64

###### Philippines has the highest counts of repayments in general, and it is also leading in irregular repayments by a very big margin.

###### Kenya has the highest weekly and monthly repayments.

###### Colombia has the highest daily repayments.

##### Kenya appears top3 in every repayment interval category. This means that irrespective of time, most Kenyans repay their loans.

In [127]:
repayment_gender = data.pivot_table(index='repayment_interval', values='loan_amount', columns='gender', aggfunc='count')
repayment_gender.sort_values(by='F', ascending=False)

gender                   F      M
repayment_interval               
monthly             250967  89204
irregular           237896  18677
daily                39027  30611
weekly                 571     31

Female customers have the highest repaymnet intervals in all categories, compared to male customers.

Monthly repayment intervals are highly done by both male and female.
This could mean that most sector activities customers are involved in are able to provide a returns on a monthly basis.

In [68]:
#which year,month were most loans taken?
#filter by year or month. How?

A temporary conclusion(some questions still unanswered):
Most Kiva borrowers are female.
Most borrowers(male and female) come from the Phillipines.
Phillipines is also the country with the highest total loan amount borrowed.
Phillipines is also the country that repaid the most loans, irregularly.
I think it's safe to say that Phillipines is an outlier.
The agriculture sector contibutes highly to most loans borrowed and repaid.
Most loan repaymnet intervals are monthly, followed closely by irregularly.
Therefore, poverty levels of Kiva borrowers are low since most people borrow loans to sustain the most basic and largest economic activity- Agriculture.